src/main/scala/ai/privado/languageEngine/java/passes/config/PropertiesFilePass.scala

```diff
@@ -23,66 +23,35 @@
-import com.github.wnameless.json.flattener.JsonFlattener
-import io.circe.yaml.parser
-import io.joern.x2cpg.SourceFiles
-import io.shiftleft.codepropertygraph.generated.nodes.{Literal, MethodParameterIn, NewFile, NewJavaProperty}
+import io.shiftleft.codepropertygraph.generated.nodes.{
+  AnnotationParameterAssign,
+  JavaProperty,
+  Literal,
+  Member,
+  MethodParameterIn
+}
 import io.shiftleft.codepropertygraph.generated.{Cpg, EdgeTypes}
 import io.shiftleft.semanticcpg.language._
-import org.slf4j.LoggerFactory
 import overflowdb.BatchedUpdate
 import overflowdb.traversal._
-
-import java.util.Properties
-import scala.jdk.CollectionConverters._
-import scala.util.{Failure, Success, Try}
-import scala.xml._
+import ai.privado.languageEngine.java.language.NodeStarters
+import io.shiftleft.passes.ForkJoinParallelCpgPass
 
 /** This pass creates a graph layer for Java `.properties` files.
   */
-class PropertiesFilePass(cpg: Cpg, projectRoot: String, ruleCache: RuleCache)
-    extends PrivadoParallelCpgPass[String](cpg) {
-
-  private val logger = LoggerFactory.getLogger(getClass)
-
-  override def generateParts(): Array[String] =
-    propertiesFiles(projectRoot, Set(".properties", ".yml", ".yaml", ".xml")).toArray
-
-  override def runOnPart(builder: DiffGraphBuilder, file: String): Unit = {
-    val fileNode      = addFileNode(file, builder)
-    val propertyNodes = addPropertyNodesAndConnectToUsers(file, builder)
-    propertyNodes.foreach(builder.addEdge(_, fileNode, EdgeTypes.SOURCE_FILE))
+class PropertiesFilePass(cpg: Cpg) extends ForkJoinParallelCpgPass[JavaProperty](cpg) {
+  override def generateParts(): Array[_ <: AnyRef] =
+    cpg.property.l.toArray.filter(pair => pair.name.nonEmpty && pair.value.nonEmpty)
+  override def runOnPart(builder: DiffGraphBuilder, property: JavaProperty): Unit = {
+    connectProperties(property, builder)
   }
 
-  private def addPropertyNodesAndConnectToUsers(
-    file: String,
-    builder: BatchedUpdate.DiffGraphBuilder
-  ): List[NewJavaProperty] = {
-    Try {
-      obtainKeyValuePairs(file, builder)
-    } match {
-      case Success(keyValuePairs) =>
-        val propertyNodes = keyValuePairs.map(addPropertyNode(_, builder))
-
-        propertyNodes.foreach(propertyNode => {
-          connectGetPropertyLiterals(propertyNode, builder)
-          connectAnnotatedParameters(propertyNode, builder)
-        })
-
-        propertyNodes
-      case Failure(exception) =>
-        logger.warn(exception.getMessage)
-        List()
-    }
+  private def connectProperties(property: JavaProperty, builder: DiffGraphBuilder): Unit = {
+    connectGetPropertyLiterals(property, builder)
+    connectAnnotatedParameters(property, builder)
   }
 
-  private def connectAnnotatedParameters(
-    propertyNode: NewJavaProperty,
-    builder: BatchedUpdate.DiffGraphBuilder
-  ): Unit = {
+  private def connectAnnotatedParameters(propertyNode: JavaProperty, builder: BatchedUpdate.DiffGraphBuilder): Unit = {
```

Comment:
earlier it was `NewJavaProperty` now `JavaProperty`?

Reply:
cpg.property gives back a list of JavaProperty.